bazel/repository.bzl

```diff
@@ -32,10 +32,10 @@ def opentelemetry_cpp_deps():
     maybe(
         http_archive,
         name = "com_github_grpc_grpc",
-        sha256 = "d6277f77e0bb922d3f6f56c0f93292bb4cfabfc3c92b31ee5ccea0e100303612",
-        strip_prefix = "grpc-1.28.0",
+        sha256 = "2060769f2d4b0d3535ba594b2ab614d7f68a492f786ab94b4318788d45e3278a",
+        strip_prefix = "grpc-1.33.2",
         urls = [
-            "https://github.com/grpc/grpc/archive/v1.28.0.tar.gz",
+            "https://github.com/grpc/grpc/archive/v1.33.2.tar.gz",
```

Comment:
Why upgrade this to 1.33? I think 1.34 is used in our CI.

https://github.com/open-telemetry/opentelemetry-cpp/blob/0f6199f9f78f11f7192bf66f28e12e80d0556164/ci/setup_grpc.sh#L16

Reply:
I think 1.33 is the only version support both bazel 4+ and legacy gcc 4.8. 
` setup_grpc.sh` is only used when using cmake, not bazel.
I found grpc 1.34 upgrade abseil-cpp to `2020923.2` and it uses something gcc 4.8 do not support.
I submit another PR #666 and try to upgrade gRPC but failed on legacy toolchain. Maybe some one can help to switch the 
grpc to old version or just disable OTLP Exporters when using bazel?I'm not familar with bazel.

Reply:
I assume for Windows, while eventually getting to prebuilt DLL / SDK, we can most likely use whatever recent version of gRPC from vcpkg? Mine right now is 1.37.0

```console
C:\work\opentelemetry-cpp\tools\vcpkg>vcpkg list grpc                                                                                                                                                                                                                           grpc:x64-windows                                   1.37.0#2         An RPC library and framework                     
```

I hope that one works with Visual Studio 2015 Update 2..
```

Reply:
I tested several toolchains with different versions of gRPC, and here is the result:

+ When using bazel to build opentelemetry-cpp
  + gRPC 1.33(which upgrade protobuf to 1.34) support bazel 4
  + gRPC 1.28-1.32 only support bazel 3
  + gRPC 1.28-1.33 support legacy gcc 4.8 while 1.34 or upper (which upgrade abseil to 20200923.X) don't
+ When using cmake to build opentelemetry-cpp
  + Just like bazel, and the official support of vcpkg is gcc 6+.
  + The protobuf (3.15) in the latest release of vcpkg(2021.05.12) do not support the latest MSVC 1929(VS 16.10)
    > Maybe we need wait for newer version to work with MSVC 1929(VS 16.10) or upper.

Right now, I just update gRPC in bazel to 1.33 for compatiability. Maybe it can be upgraded to newer version in the future and in another PR.